### PR TITLE
Added PyInterpreterTool

### PR DESCRIPTION
**Pull request by mishushakov**
I have added the Pyodide interpreter to LangChainJS

Example:

```ts
import { StringOutputParser } from "../../schema/output_parser.js";
import { OpenAI } from "../../llms/openai.js";
import { PromptTemplate } from "../../prompts/index.js";
import { PythonInterpreterTool } from "../pyinterpreter.js";

const prompt = PromptTemplate.fromTemplate(
  `Can you generate python code that: {input}?`
);

const model = new OpenAI({});

const interpreter = new PythonInterpreterTool({
  indexURL: "../node_modules/pyodide",
});

const chain = prompt
  .pipe(model)
  .pipe(new StringOutputParser())
  .pipe(interpreter);

const result = await chain.invoke({
  input: "print the first 10 numbers of the fibonacci sequence",
});

console.log(result)

// prints:
// {"stdout": "0112358132134", "stderr": ""}
```

### CODE REVIEW

File: langchain/scripts/create-entrypoints.js [hunk 1]
```diff
@@ -326,6 +326,7 @@ const requiresOptionalDependency = [
   "tools/sql",
   "tools/webbrowser",
   "tools/google_calendar",
+  "tools/pyinterpreter",
```
Comment (jacoblee93) [inline langchain/scripts/create-entrypoints.js:329]: Let's move to `experimental` since it's potentially running arbitrary code

Comment (jacoblee93) [reply langchain/scripts/create-entrypoints.js:329]: You'll also need to add an entrypoint above

File: langchain/package.json [hunk 2]
```diff
@@ -1011,6 +1012,7 @@
     "playwright": "^1.32.1",
     "portkey-ai": "^0.1.11",
     "puppeteer": "^19.7.2",
+    "pyodide": "0.24.1",
```
Comment (jacoblee93) [inline langchain/package.json:1015]: Add under `peerDependenciesMeta` below

File: langchain/src/tools/index.ts [hunk 3]
```diff
@@ -44,3 +44,4 @@ export {
 export { SearxngSearch } from "./searxng_search.js";
 export { SearchApi, type SearchApiParameters } from "./searchapi.js";
 export { formatToOpenAIFunction } from "./convert_to_openai.js";
+export { PythonInterpreterTool } from "./pyinterpreter.js";
```
Comment (jacoblee93) [inline langchain/src/tools/index.ts:47]: We can't export from here if it requires an optional peer dep, we'll need to use a custom entrypoint

**Review by jacoblee93 — CHANGES_REQUESTED** [review]
Very cool, needs docs as well. Can you also add a security warning to those as well?

**Comment by mishushakov** [comment]
Of course. Where should I put them? In the comments or do you have a special place for tools anywhere? 😄

What should the warning say? The code is executed in secure WASM sandbox...

I also have a question regarding the tests. Should it only test the tool itself or the complete chain like I did there?

Thanks

**Comment by jacoblee93** [comment]
> Of course. Where should I put them? In the comments or do you have a special place for tools anywhere? 😄
> 
> What should the warning say? The code is executed in secure WASM sandbox...
> 
> I also have a question regarding the tests. Should it only test the tool itself or the complete chain like I did there?
> 
> Thanks

Best place would be to add a new folder called `tools/experimental` and put it there.

**Comment by mishushakov** [comment]
Ok done. Let me know where I should put the docs and the warning

**Comment by jacoblee93** [comment]
Hi so sorry, it's been a crazy week for us - will look at this and write docs tomorrow!

**Comment by jacoblee93** [comment]
Hey @mishushakov, I moved things around but the integration test seems rather flaky - do you think you could come up with a more consistent one?

**Comment by mishushakov** [comment]
Hey, I have been busy this week, but happy to tackle this on the weekend!

**Comment by mishushakov** [comment]
Is this better? Or are you looking for something else entirely?

```
import { test, expect } from "@jest/globals";
import { StringOutputParser } from "../../../schema/output_parser.js";
import { OpenAI } from "../../../llms/openai.js";
import { PromptTemplate } from "../../../prompts/index.js";
import { PythonInterpreterTool } from "../pyinterpreter.js";

describe("Python Interpreter testsuite", () => {
  test("hello langchain", async () => {
    const prompt = PromptTemplate.fromTemplate(
      `Can you generate python code that: {input}?`
    );

    const model = new OpenAI({});

    const interpreter = await PythonInterpreterTool.initialize({
      indexURL: "../node_modules/pyodide",
    });
    const chain = prompt
      .pipe(model)
      .pipe(new StringOutputParser())
      .pipe(interpreter);

    const result = await chain.invoke({
      input: `prints "Hello LangChain"`,
    });

    expect(JSON.parse(result).stdout).toBe("Hello LangChain");
  });
});
```

**Comment by mishushakov** [comment]
The test does require OpenAI API key though

**Comment by mishushakov** [comment]
Otherwise, a very basic test, that only tests the tool itself would look like:

```ts
import { test, expect } from "@jest/globals";
import { PythonInterpreterTool } from "../pyinterpreter.js";

describe("Python Interpreter testsuite", () => {
  test("hello langchain", async () => {
    const interpreter = await PythonInterpreterTool.initialize({
      indexURL: "../node_modules/pyodide",
    });

    const result = await interpreter._call(`print("Hello LangChain")`)
    expect(JSON.parse(result).stdout).toBe("Hello LangChain");
  });
});
```

**Review by jacoblee93 — APPROVED** [review]

**Comment by jacoblee93** [comment]
Thank you! It's really neat

**Pull request merged**